R/mincolumns.R

```diff
@@ -0,0 +1,261 @@
+{
+    extremeColumns(x,
+               remove.missing = remove.missing,
+               remove.rows = remove.rows,
+               remove.columns = remove.columns,
+               subset = subset,
+               warn = warn,
+               function.name = sQuote(deparse(sys.call()[[1]])))
+}
+
+#' @rdname ExtremeOperations
+#' @export
+MinColumns <- MinEachColumn
+
+#' @rdname ExtremeOperations
+#' @examples
+#'
+#' input.matrix <- matrix(1:8, nrow = 4)
+#' MaxEachColumn(input.matrix) #= c(10, 26)
+#' named.matrix <- input.matrix
+#' dimnames(named.matrix) <- list(LETTERS[1:4], letters[1:2])
+#' MaxEachColumn(named.matrix)
+#' MaxEachColumn(named.matrix, remove.rows = c("A", "C"))
+#' MaxEachColumn(named.matrix, subset = c(TRUE, FALSE, TRUE, FALSE))
+#' MaxEachColumn(named.matrix, remove.rows = c("B", "D"))
+#' @export
+MaxEachColumn <- function(x,
+                          remove.missing = TRUE,
+                          remove.rows = c("NET", "SUM", "Total"),
+                          remove.columns = NULL,
+                          subset = NULL,
+                          warn = FALSE)
+{
+    extremeColumns(x,
+               remove.missing = remove.missing,
+               remove.rows = remove.rows,
+               remove.columns = remove.columns,
+               subset = subset,
+               warn = warn,
+               function.name = sQuote(deparse(sys.call()[[1]])))
+}
+
+#' @rdname ExtremeOperations
+#' @export
+MaxColumns <- MaxEachColumn
+
+extremeColumns <- function(x,
+                       remove.missing = TRUE,
+                       remove.rows = c("NET", "SUM", "Total"),
+                       remove.columns = NULL,
+                       subset = NULL,
+                       warn = FALSE,
+                       function.name)
+{
+    x <- processArguments(list(x),
+                          remove.missing = remove.missing,
+                          remove.rows = remove.rows, remove.columns = remove.columns,
+                          subset = NULL, weights = NULL,
+                          check.statistics = FALSE,
+                          warn = warn,
+                          function.name = function.name)
+    input <- subsetAndWeightInputsIfNecessary(x,
+                                              subset = subset, weights = NULL,
+                                              return.total.element.weights = FALSE,
+                                              warn = warn,
+                                              function.name = function.name)[[1L]]
+    output <- extremeCols(input, function.name, remove.missing = remove.missing)
+    if (warn && NROW(input) == 1L)
+            throwWarningAboutCalculationWithSingleElement(input, dimension = 1L, function.name)
+
+    return(output)
+}
+
+#' @importFrom stats setNames
+#' @noRd
+extremeCols <- function(x, function.name, remove.missing = TRUE, dims)
+{
+    extremum.fun <- ifelse(grepl("Max", function.name), max, min)
```

Comment:
Minor speed improvement `startsWith(function.name, "Max")`

Reply:
Not possible in this case because `function.name` is actually `sQuote("Max")` or `sQuote("Min")`.